### DISCUSSION TITLE
Cloning an API or extending from a template

### DISCUSSION THREAD

**Discussion by kaloctemha** [ApicurioArchive/apicurio-studio#2194] [Q&A] [answered]
Hi all,

I am going to create a bunch of APIs through apicurio-studio. All of the APIs have the same configurations such as Info, Contact, Servers, Security Schemes and Requirements etc. I would like to create a template API having all of those configured and derive all the APIs from that so that I only need to change the Operations and create Data Types Paths and Responses etc. Or just duplicate an already created API and modify accordingly.

Cheers,
Ahmet

**Comment by EricWittmann** [comment, marked as answer]
We had aspirations to allow users to provide their own templates.  There is some work done in this area already, but it's not exposed via the UI and it's not usable currently in the cloud deployment of Studio.  If you're running your own instance of Studio you could probably make your own templates.  But if you're using the one we run in the cloud (which is only for evaluation purposes!) then your only option is to Clone an existing API.

Here is the API for templates:

https://github.com/Apicurio/apicurio-studio/blob/master/back-end/hub-api/src/main/java/io/apicurio/hub/api/rest/ITemplatesResource.java

**Reply by kaloctemha** [reply]
Hi Eric, my organization currently only running on the cloud so I will go with the Cloning option. Thanks a lot for the details
